base_domain_operator/README.rst

```diff
@@ -0,0 +1,77 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+=======================
+Custom domain operators
+=======================
+
+This module was written to allow developers to define custom domain operators. This can be very helpful for complicated selections or ir.rules.
+
+Currently implemented operators
+===============================
+
+``parent_of``
+  The inverse of ``child_of``: Searches for the parents of given records in the hierarchy, including the records themselves.
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. depend on it
+#. use one of the operators explained above in any of your domains
+
+To add your own operators:
+
+#. in ``base.domain.operator``, define a function ``_operator_${your_operator}`` with decorator ``@api.model`` and signature ``leaf, expression``
+#. ``leaf`` is a 3-tuple of a domain proposition, ``expression`` an instance of ``openerp.osv.expression.expression``
+#. return a list of ``openerp.osv.expression.ExtendedLeaf`` instances that maps your operator to some expressions the original domain parser can evaluate
+#. note that you can use the internal ``inselect`` operator here if you pass ``internal=True`` to ``ExtendedLeaf``'s constructor
+#. be careful with using the ORM in those handlers, you're busy modifying a crucial part of the ORM
+#. take good care that you don't introduce SQL injections and other security problems
+
+Known issues / Roadmap
+======================
+
+* given the upstream code is not exactly extension friendly, we'll have to reimplement a bunch of helper functions
+* ``parent_of`` currently only supports the most basic use cases, this needs to be amended as needed
```

Comment:
What do you refer about basic use cases? Can it be specified? I suppose for example that dot notation is not included.